src/VisualStudio/Core/Def/Progression/GraphProvider.cs

```diff
@@ -162,15 +164,12 @@ public void BeginGetGraphData(IGraphContext context)
 
             var graphQueries = GetGraphQueries(context, _threadingContext, _asyncListener);
 
-            if (graphQueries.Count > 0)
-            {
-                _graphQueryManager.AddQueries(context, graphQueries);
-            }
-            else
-            {
-                // It's an unknown query type, so we're done
-                context.OnCompleted();
-            }
+            // Perform the queries asynchronously  in a fire-and-forget fashion.  This helper will be responsible
+            // for always completing the context.
+            var asyncToken = _asyncListener.BeginAsyncOperation(nameof(BeginGetGraphData));
+            _ = _graphQueryManager
+                .AddQueriesAsync(context, graphQueries)
```

Comment:
Can AddQueriesAsync accidentally throw synchronously, and we'd never call CompletesAsyncOperation? I think potentially yes, if the query would get a triggered cancellation.

Reply:
no.  it can't.  it's an 'async' method so it always returns a task.